rcl/src/rcl/rmw_implementation_identifier_check.c

```diff
@@ -70,14 +71,14 @@ INITIALIZER(initialize) {
       ROS_PACKAGE_NAME,
       "Error getting env var '" RCUTILS_STRINGIFY(RMW_IMPLEMENTATION_ENV_VAR_NAME) "': %s\n",
       get_env_error_str);
```

Comment:
@Blast545 meta: why not using `RCL_SET_ERROR_MSG_WITH_FORMAT` and logging in the caller? I'd think that'd make the test less noisy.

Reply:
What do you mean noisy in the context? isn't the check logging only in the error condition? (if any)

Reply:
Correct, error conditions that we will test for. It's even easier to assert on error messages if we defer logging.

Reply:
Addressed with 9931857. Although `RCUTILS_LOG_ERROR_NAMED(ROS_PACKAGE_NAME, rcl_get_error_string().str);` is throwing me this warning 
```
/home/misato/ros_ws/src/ros2/rcl/rcl/include/rcl/error_handling.h:45:30: warning: format not a string literal and no format arguments [-Wformat-security]
 #define rcl_get_error_string rcutils_get_error_string
                              ^
/home/misato/ros_ws/src/ros2/rcl/rcl/src/rcl/rmw_implementation_identifier_check.c:158:47: note: in expansion of macro ‘rcl_get_error_string’
     RCUTILS_LOG_ERROR_NAMED(ROS_PACKAGE_NAME, rcl_get_error_string().str);
```

Reply:
Solved already